Heading component type declaration not recognise by typescript compiler

### Current behaviour

I found an issue in the heading component when I try to use the 107.0.0 version
Our current code is

`<Heading title={t("my translation sample ")} divider={false} />`

```
The npm run lint command returns now the following error
error TS2740: Type '{ title: any; divider: false; }' is missing the following properties from type '{ [x: string]: any; children: any; backLink: any; divider?: boolean | undefined; help: any; helpAriaLabel: any; helpLink: any; pills: any; separator?: boolean | undefined; subheader: any; subtitleId: any; title: any; titleId: any; }': children, backLink, help, helpAriaLabel, and 5 more.
```

If I set the non optional properties in the code, the error is not displayed after executing the command npm run lint
```
        <Heading
          title={t("my translation sample ")}
          divider={false}
          subheader="test"
          backLink="test"
          help="test"
          helpAriaLabel="test"
          helpLink="test"
          pills="test"
          subtitleId="test"
          titleId="test"
        >
          <></>
        </Heading>
```


### Expected behaviour

no error return on npm run lint

### CodeSandbox or Storybook URL

No code sandbox

### JIRA Ticket (Sage Only)

_No response_

### Suggested Solution

_No response_

### Carbon Version

107.0.0

### Design Tokens Version

_No response_

### What browsers are you seeing the problem on?

Other

### What Operating System are you seeing the problem on?

Windows

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

--- a/src/components/heading/index.d.ts
+++ b/src/components/heading/index.d.ts
@@ -1,1 +1,1 @@
-export { default } from "./heading.component";
+export { default } from "./heading";

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -22,6 +22,5 @@
     "strict": true,
     "target": "es6"
   },
-  "include": ["src/**/*"],
-  "exclude": ["src/**/*.stories.js", "src/**/*.test.js"]
+  "include": ["./src/**/*.ts*"],
 }

